gprofiler/python.py

```diff
@@ -188,35 +199,26 @@ def start(self):
             str(self._frequency),
             # Duration is irrelevant here, we want to run continuously.
         ]
-        process = start_process(cmd)
-        try:
-            poll_process(process, self.poll_timeout, self._stop_event)
-        except TimeoutError:
-            # Process is alive. So far, so good...
-            if not self.output_path.exists():
-                process.kill()
-                stdout = process.stdout.read().decode()
-                stderr = process.stderr.read().decode()
-                self._check_missing_headers(stdout)
-                raise CalledProcessError(process.returncode, process.args, stdout, stderr)
-            # happy flow
-            self.process = process
-        else:
-            # process terminated, must be an error:
-            self._check_output(process, self.output_path)
-            raise RuntimeError("_check_output did not raise!")
+        self.process = start_process(cmd)
+        # no need to check for success here - we're already past calling test(), PyPerf has been already proved
+        # working.
 
     def _dump(self) -> Path:
         assert self.process is not None, "profiling not started!"
         self.process.send_signal(self.dump_signal)
         # important to not grab the transient data file
-        while True:
+        for _ in range(self.dump_poll_retries):
             output_files = glob.glob(f"{str(self.output_path)}.*")
             if output_files:
                 break
 
-            if self._stop_event.wait(0.1):
+            if self._stop_event.wait(self.dump_poll_interval):
                 raise StopEventSetException()
+        else:
+            if self.process.poll() is not None:
+                self._pyperf_error(self.process)
+            else:
+                raise Exception("PyPerf is not responding!")
```

Comment:
Then kill it?

Reply:
I'm not sure what to do here. We can kill it, and what? Restart it? I don't wanna get into a restart loop of PyPerf due to an error.

At this point I want someone to inspect the exception & PyPerf's output (perhaps I can add the output here then)

Reply:
No, just kill it and revert to py-spy

Reply:
Sounds like a plan

Reply:
Done